Src/EventCounterCollector/EventCounterCollector/EventCounterCollectionModule.cs

```diff
@@ -0,0 +1,111 @@
+﻿namespace Microsoft.ApplicationInsights.Extensibility.EventCounterCollector
+{
+    using System;
+    using System.Collections;
+    using System.Collections.Generic;
+    using System.Globalization;
+    using System.Text;
+    using Microsoft.ApplicationInsights;
+    using Microsoft.ApplicationInsights.Common;
+    using Microsoft.ApplicationInsights.Extensibility.EventCounterCollector.Implementation;
+    using Microsoft.ApplicationInsights.Extensibility.Implementation;
+
+    /// <summary>
+    /// Telemetry module for collecting EventCounters.
+    /// https://github.com/dotnet/corefx/blob/master/src/System.Diagnostics.Tracing/documentation/EventCounterTutorial.md.
+    /// </summary>
+    public class EventCounterCollectionModule : ITelemetryModule, IDisposable
+    {
+        private readonly double refreshInternalInSecs = 60;
+
+        /// <summary>
+        /// TelemetryClient used to send data.
+        /// </summary>        
+        private TelemetryClient client = null;        
+        private EventCounterListener eventCounterListener;
+        private bool disposed = false;
+        private bool isInitialized = false;        
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="EventCounterCollectionModule"/> class.
+        /// </summary>
+        public EventCounterCollectionModule()
+        {
+            this.Counters = new List<EventCounterCollectionRequest>();
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="EventCounterCollectionModule"/> class.
+        /// </summary>
+        internal EventCounterCollectionModule(double refreshIntervalInSecs) : this()
+        {
+            this.refreshInternalInSecs = refreshIntervalInSecs;
+        }
+
+        /// <summary>
+        /// Gets the list of counter names to collect. Each should have the name of EventSource publishing the counter, and counter name.
+        /// </summary>
+        public IList<EventCounterCollectionRequest> Counters { get; private set; }
+
+        /// <summary>
+        /// IDisposable implementation.
+        /// </summary>
+        public void Dispose()
+        {
+            this.Dispose(true);
+            GC.SuppressFinalize(this);
+        }        
+
+        /// <summary>
+        /// Initialize method is called after all configuration properties have been loaded from the configuration.
+        /// </summary>
+        public void Initialize(TelemetryConfiguration configuration)
+        {
+            try
+            {                
+                if (!this.isInitialized)
+                {
+                    EventCounterCollectorEventSource.Log.ModuleIsBeingInitializedEvent(string.Format(
```

Comment:
when using EventSource, it is better to format string internally (inside the event). In the majority of cases (there is no listener), event is not actually written and there is no need to allocate extra string. Not very important here, but still best to follow best practices

Reply:
Also event source formats this message for you and passes args alone